server/src/main/java/com/hedera/block/server/persistence/storage/read/BlockAsLocalFileReader.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.block.server.persistence.storage.read;
+
+import com.hedera.block.common.utils.Preconditions;
+import com.hedera.hapi.block.BlockUnparsed;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Optional;
+
+/**
+ * A Block reader that reads block-as-file.
+ */
+public final class BlockAsLocalFileReader implements LocalBlockReader<BlockUnparsed> {
+    /**
+     * Constructor.
+     */
+    private BlockAsLocalFileReader() {}
+
+    /**
+     * This method creates and returns a new instance of
+     * {@link BlockAsLocalFileReader}.
+     *
+     * @return a new, fully initialized instance of
+     * {@link BlockAsLocalFileReader}
+     */
+    public static BlockAsLocalFileReader newInstance() {
```

Comment:
Yep, you are absolutely correct. The reader _needs_ a path resolver, and yes, I am not there yet, this will be the follow up PR after this one, which will address the follow up complementary logic for reading a block as a local file #349. I have proactively created the underlying classes for readers/writers/resolvers/removers in order to not make dagger blow up or change it to allow nulls (I think that is a bad idea) and since I will need them literally in the follow up issues. For simplicity sake, the above mentioned are currently void of any logic or dependencies.